### PR TITLE
🎉 New source: Qualaroo

### PR DESCRIPTION
**Pull request by danieldiamond**
## What
Adding a new source: [Qualaroo](https://qualaroo.com/)

## How
See PR

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
@danieldiamond  thanks for the contribution. We will review this soon.

**Comment by danieldiamond** [comment]
```
 python -m pytest unit_tests
Test session starts (platform: darwin, Python 3.9.9, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/danieldiamond/repos/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, requests-mock-1.9.3, mock-3.6.1, timeout-1.4.2
collecting ...
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_request_params ✓              25% ██▌
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_next_page_token ✓             50% █████
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_surveys_stream ✓              75% ███████▌
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_responses_stream ✓           100% ██████████
====================================================== warnings summary ======================================================
airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py: 11 warnings
  /Users/danieldiamond/.virtualenvs/qualaroo/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:38: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator = NoAuth()

airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py: 11 warnings
  /Users/danieldiamond/.virtualenvs/qualaroo/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (0.20s):
       4 passed
```

File: airbyte-integrations/connectors/source-qualaroo/source_qualaroo/schemas/responses.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        "string"
+      ]
+    },
+    "answered_questions": {
+      "type": [
+        "null",
+        "object"
+      ],
+      "properties": {
+        "581047": {
```
Comment (danieldiamond) [inline airbyte-integrations/connectors/source-qualaroo/source_qualaroo/schemas/responses.json:73]: these `answered_questions` dynamic keys need to be addressed prior to merging

**Comment by marcosmarxm** [comment]
@danieldiamond can you post the integratoin test results too?

**Comment by danieldiamond** [comment]
```
➜ python -m pytest unit_tests
Test session starts (platform: darwin, Python 3.9.9, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/danieldiamond/repos/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, requests-mock-1.9.3, mock-3.6.1, timeout-1.4.2
collecting ...
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_request_params ✓                       25% ██▌
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_next_page_token ✓                      50% █████
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_surveys_stream ✓                       75% ███████▌
 airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py::test_responses_stream ✓                    100% ██████████
========================================================== warnings summary ===========================================================
airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py: 11 warnings
  /Users/danieldiamond/.virtualenvs/qualaroo/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:38: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator = NoAuth()

airbyte-integrations/connectors/source-qualaroo/unit_tests/test_streams.py: 11 warnings
  /Users/danieldiamond/.virtualenvs/qualaroo/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (0.39s):
       4 passed
```

**Comment by marcosmarxm** [comment]
@danieldiamond can you run `./gradlew airbyte-integrations:connectors:source-qualaroo:integrationTest`?

**Comment by danieldiamond** [comment]
@marcosmarxm misread your previous comment. here's the integration tests:

```
> Task :airbyte-integrations:connectors:source-qualaroo:sourceAcceptanceTest
============================= test session starts ==============================
platform linux -- Python 3.7.11, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /test_input
plugins: sugar-0.9.4, timeout-1.4.2
collected 13 items / 1 skipped / 12 selected

test_core.py ............                                                [ 92%]
test_full_refresh.py .                                                   [100%]

=========================== short test summary info ============================
SKIPPED [1] ../usr/local/lib/python3.7/site-packages/source_acceptance_test/plugin.py:56: Skipping TestIncremental.test_two_sequential_reads because not found in the config
======================== 13 passed, 1 skipped in 21.76s ========================

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.2/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 1 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD SUCCESSFUL in 47s
81 actionable tasks: 44 executed, 37 up-to-date
```

**Comment by danieldiamond** [comment]
@marcosmarxm do you need anything else on my end to progress this?

**Comment by harshithmullapudi** [comment]
@marcosmarxm can I take this up?

**Comment by marcosmarxm** [comment]
I requested the credentials in https://github.com/airbytehq/airbyte/issues/8800. @danieldiamond do you have tests credentials to share to apply in tests?

**Comment by danieldiamond** [comment]
unfortunately i dont have test credentials

File: airbyte-integrations/connectors/source-qualaroo/bin/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+*
+!Dockerfile
+!Dockerfile.test
+!main.py
+!source_qualaroo
+!setup.py
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-qualaroo/bin/.dockerignore:6]: are these files duplicated ?

File: airbyte-integrations/connectors/source-qualaroo/bin/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# build and load all requirements
+FROM base as builder
+WORKDIR /airbyte/integration_code
+
+# upgrade pip to the latest version
+RUN apk --no-cache upgrade \
+    && pip install --upgrade pip
+
+COPY setup.py ./
+# install necessary packages to a temporary folder
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-qualaroo/bin/Dockerfile:12]: are these files duplicated ? Looks like bin folder is not needed

File: airbyte-integrations/connectors/source-qualaroo/bin/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+* required for your connector to work need to go to `MAIN_REQUIREMENTS` list.
+* required for the testing need to go to `TEST_REQUIREMENTS` list
+
+### Publishing a new version of the connector
+You've checked out the repo, implemented a million dollar feature, and you're ready to share your changes with the world. Now what?
+1. Make sure your changes are passing unit and integration tests.
+1. Bump the connector version in `Dockerfile` -- just increment the value of the `LABEL io.airbyte.version` appropriately (we use [SemVer](https://semver.org/)).
+1. Create a Pull Request.
+1. Pat yourself on the back for being an awesome contributor.
+1. Someone from Airbyte will take a look at your PR and iterate with you to merge it into master.
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-qualaroo/bin/README.md:131]: are these files duplicated ?

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]
I think we can delete bin folder

File: airbyte-integrations/connectors/source-qualaroo/bin/.dockerignore (hunk 2 above)
Comment (danieldiamond) [reply airbyte-integrations/connectors/source-qualaroo/bin/.dockerignore]: yeah used another connector as boilerplate, forgot to remove

File: airbyte-integrations/connectors/source-qualaroo/bin/Dockerfile (hunk 3 above)
Comment (danieldiamond) [reply airbyte-integrations/connectors/source-qualaroo/bin/Dockerfile]: yeah used another connector as boilerplate, forgot to remove

File: airbyte-integrations/connectors/source-qualaroo/bin/README.md (hunk 4 above)
Comment (danieldiamond) [reply airbyte-integrations/connectors/source-qualaroo/bin/README.md]: yeah used another connector as boilerplate, forgot to remove

**Comment by danieldiamond** [comment]
@harshithmullapudi have you received your credentials for the test qualaroo account?

**Comment by marcosmarxm** [comment]
> @harshithmullapudi have you received your credentials for the test qualaroo account?

yes.

**Pull request merged**